packages/vite/src/node/utils.ts

```diff
@@ -21,20 +21,26 @@ import type { TransformResult } from 'rollup'
 import { createFilter as _createFilter } from '@rollup/pluginutils'
 import {
   CLIENT_ENTRY,
-  CLIENT_PUBLIC_PATH,
   DEFAULT_EXTENSIONS,
-  ENV_PUBLIC_PATH,
   FS_PREFIX,
   NULL_BYTE_PLACEHOLDER,
   OPTIMIZABLE_ENTRY_RE,
   VALID_ID_PREFIX,
   loopbackHosts,
   wildcardHosts,
 } from './constants'
+import {
+  cleanUrl,
+  isWindows,
+  normalizePath,
+  removeTimestampQuery,
+} from './utils/shared'
 import type { DepOptimizationConfig } from './optimizer'
 import type { ResolvedConfig } from './config'
 import type { ResolvedServerUrls } from './server'
-import type { CommonServerOptions } from '.'
+import type { CommonServerOptions } from './index'
+
+export * from './utils/shared'
```

Comment:
I'm not really sure of extracting shared utils to a separate file, that changes the git history a lot. Is there a way to configure Rollup to chunk in a way that reduces the imports, when importing `./runtime/client`?

Reply:
What is a lot to you? I specifically made re export to not rewrite other imports.

I don't think it's possible to extract only a subset of functions in rollup, but I didn't investigate it.

Reply:
A lot of the utilities has attached git history to them that helps with debugging things, moving it to another file kinda cuts off that history, making it harder to debug. You could still manually navigate to the old history, but this would make things harder.

Reply:
We can add a last hash to the file before it was moved, so it's easier to find the previous version 🤷🏻 

I don't really think this is a blocker, the history is still there, but it takes one extra step to get to it, which is like 5 seconds.

Reply:
Alright, looking back I think it's not as bad, so probably fine with it for now if it's needed.